test/unit/org/apache/cassandra/auth/AuthCacheTest.java

```diff
@@ -322,6 +322,41 @@ public void testCacheLoaderIsNotCalledOnGetAllWhenCacheIsNotEmpty()
         assertEquals(1, loadCounter);
     }
 
+    @Test
+    public void testMetricsOnCacheEnabled()
+    {
+        TestCache authCache = new TestCache(this::countingLoader, this::emptyBulkLoader, this::setValidity, () -> validity, () -> isCacheEnabled);
+        authCache.get("10");
+        authCache.get("11");
+
+        assertThat(authCache.getMetrics().requests.getCount()).isEqualTo(2L);
+        assertThat(authCache.getMetrics().hits.getCount()).isEqualTo(0L);
+        assertThat(authCache.getMetrics().misses.getCount()).isEqualTo(2L);
+        assertEquals(2, loadCounter);
+
+        authCache.get("10");
+        authCache.get("11");
+
+        assertThat(authCache.getMetrics().requests.getCount()).isEqualTo(4L);
+        assertThat(authCache.getMetrics().hits.getCount()).isEqualTo(2L);
+        assertThat(authCache.getMetrics().misses.getCount()).isEqualTo(2L);
+        assertEquals(2, loadCounter);
+    }
+
+    @Test
+    public void testMetricsOnCacheDisabled()
```

Comment:
Should we add a test to check that new metrics are available via the JMX API? I haven't found a good candidate for a test class, but I think it's still worth checking that `getMaxEntries` and `setMaxEntries` match, and the `entries` match the number entries we set to the cache.

Reply:
I dont think that involving _JMX_ is necessary. We basically just need to invoke same MBean method programmatically. I have added "testSettingMaxEntries" test for that.